### DISCUSSION TITLE
ScrollContainer and WebView

### DISCUSSION THREAD

**Discussion by wanglifeng95** [beeware/toga#1966] [Q&A]
1、Use of ScrollContainer. It seems that the scroll box cannot display all the sub items through scrolling, only a portion can be seen.
2、Use of WebView. Is the on_key_down event in WebView triggered when a button is clicked or when there is input from the keyboard in WebView? I have tried Toga_example ,and it seems that neither of the above two methods can activate this event, perhaps I did not understand it correctly. Can you provide a simple example of this event? I don't quite understand how to use on_key_down event.

**Comment by freakboy3742** [comment]
Regarding point 1 - you'll need to provide an example. "It seems" isn't enough detail for us to use for debugging purposes.

Regarding point 2 - The `on_key_down` handler was intended as a keyboard handler; however, it is scheduled to be removed from WebView in the very near future (see #1949), so I wouldn't spend any time trying to understand it.

**Reply by wanglifeng95** [reply]
> Regarding point 1 - you'll need to provide an example. "It seems" isn't enough detail for us to use for debugging purposes.
> 
> Regarding point 2 - The `on_key_down` handler was intended as a keyboard handler; however, it is scheduled to be removed from WebView in the very near future (see #1949), so I wouldn't spend any time trying to understand it.

The ScrollContainer example:

import toga
from toga.style.pack import Pack, COLUMN

class Quantity(toga.App):

    def startup(self):
        self.count=1
        self.box=toga.Box(style=Pack(direction=COLUMN))
        main_box = toga.Box(style=Pack(direction=COLUMN))
        for n in range(0,100):
            label = toga.Label(f"label {n}")
            self.box.add(label)
        main_scroll=toga.ScrollContainer(style=Pack(flex=1))
        main_scroll.content=self.box
        addWidgetsBut=toga.Button("Add Label",on_press=self.butEvent)
        main_box.add(main_scroll)
        main_box.add(addWidgetsBut)
        self.main_window = toga.MainWindow(title=self.formal_name)
        self.main_window.content = main_box
        self.main_window.show()

    def butEvent(self,widget):
        print(f"New label {self.count}")
        label=toga.Label(f"New label {self.count}")
        self.box.add(label)
        self.count+=1

def main():
    return Quantity()


Added 100 labels to the Scroll Box, but in reality, only less than half of the labels can be displayed. I don't know why?

**Reply by wanglifeng95** [reply]
I found that the issue with ScrollContainer only exists in Windows, but it can be displayed normally in Ubuntu

**Reply by freakboy3742** [reply]
Sounds like you've found a bug in the Windows implementation of Scrollview. I'm in the middle of building GUI tests for this widget; I should be looking at the Windows implementation next week.

**Reply by wanglifeng95** [reply]
Thanks for your efforts. Hope this bug can be fixed later
